### PR TITLE
Experimental/psr12 pipeline check

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/ADF-1161

# Goal

- Adds PSR-12 analysis for touched PHP files

# Example

- PR https://github.com/oat-sa/extension-tao-test/pull/434
- Action: https://github.com/oat-sa/extension-tao-test/actions/runs/3127922994/jobs/5075113530

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -62,3 +62,29 @@ runs:
       run: vendor/bin/phpunit --bootstrap bootstrap.php${{ inputs.coverage && ' --coverage-clover coverage.xml' }} "src/${{ inputs.test-suites-path }}"
     - uses: codecov/codecov-action@v3
       if: ${{ inputs.coverage }}
+    - uses: jitterbit/get-changed-files@v1
```
Comment (gabrielfs7) [inline action.yml:65]: We need to install codesniffer in the line 50

File: action.yml [hunk 2]
```diff
@@ -62,3 +62,29 @@ runs:
...
+        if [ -s "${FILELIST}" ]; then
+            vendor/bin/phpcs --standard=PSR12 --exclude=PSR1.Classes.ClassDeclaration -v --file-list="${FILELIST}"
+            RETVAL=$?
+            echo "The following files were tested for PSR-12 errors:"; echo 
+            cat -n "$FILELIST"
+        else
+            echo "There are no changes in PHP files (no PSR-12 test performed)"
+        fi
+        rm "$FILELIST"
+        exit $RETVAL
```
Comment (gabrielfs7) [inline action.yml:90]: we need to add a blank line at the end

**Review by hectoras — APPROVED** [review]
- [x] Pull request title and description are meaningful
- [x] Target branch is correct
- [ ] Commits are following conventional commits
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Kitchen links are not exposed in the description
- [x] No unnecessary dependencies are installed

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
lgtm

**Review by jalizada87 — APPROVED** [review]

**Review by jalizada87 — APPROVED** [review]

**Pull request merged**